stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/viewholder/decorator/BackgroundDecorator.kt

```diff
@@ -1,57 +1,85 @@
 package io.getstream.chat.android.ui.messages.adapter.viewholder.decorator
 
+import android.graphics.Paint
+import android.view.View
 import androidx.core.content.ContextCompat
 import com.getstream.sdk.chat.adapter.MessageListItem
+import com.google.android.material.shape.MaterialShapeDrawable
+import com.google.android.material.shape.ShapeAppearanceModel
 import io.getstream.chat.android.ui.R
 import io.getstream.chat.android.ui.messages.adapter.viewholder.MessageDeletedViewHolder
 import io.getstream.chat.android.ui.messages.adapter.viewholder.MessagePlainTextViewHolder
+import io.getstream.chat.android.ui.messages.adapter.viewholder.OnlyMediaAttachmentsViewHolder
 import io.getstream.chat.android.ui.utils.extensions.dpToPxPrecise
 
 internal class BackgroundDecorator : BaseDecorator() {
 
     override fun decorateDeletedMessage(viewHolder: MessageDeletedViewHolder, data: MessageListItem.MessageItem) {
-        val radius = DEFAULT_CORNER_RADIUS_DP.dpToPxPrecise()
-        viewHolder.binding.deleteLabel.background = BackgroundDrawable(
-            color = ContextCompat.getColor(viewHolder.itemView.context, MESSAGE_DELETED_BACKGROUND),
-            topLeftCornerPx = radius,
-            topRightCornerPx = radius,
-            bottomRightCornerPx = if (data.positions.contains(MessageListItem.Position.BOTTOM)) 0f else radius,
-            bottomLeftCornerPx = radius
-        )
+        val bottomRightCorner =
+            if (data.positions.contains(MessageListItem.Position.BOTTOM)) 0f else DEFAULT_CORNER_RADIUS
+        val shapeAppearanceModel = ShapeAppearanceModel.builder().setAllCornerSizes(DEFAULT_CORNER_RADIUS)
+            .setBottomRightCornerSize(bottomRightCorner).build()
```

Comment:
This API looks awesome 👍

Reply:
Yeah, I discovered it accidentally and was pleased  much